### PR TITLE
Subplot table

### PR DESCRIPTION
**Pull request by Kully**

### CODE REVIEW

**Comment by Kully** [comment]
fixes: https://github.com/plotly/documentation/issues/849

@cldougl ready for a review

Should the doc go in `https://plot.ly/python/subplot-charts/`? Not sure between there and the Basic Charts.

**Comment by cldougl** [comment]
yep - https://plot.ly/python/subplot-charts/

File: _posts/python/basic/table-subplots/2015-06-30-table-subplots.html [hunk 1]
```diff
@@ -0,0 +1,544 @@
+---
+permalink: python/table-subplots/
+description: How to place tables in subplot layouts in Python with Plotly.
```
Comment (cldougl) [inline _posts/python/basic/table-subplots/2015-06-30-table-subplots.html]: How to create a subplot with tables and charts in Python with Plotly.

File: _posts/python/basic/table-subplots/2015-06-30-table-subplots.html [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+name: Python Tables in Subplots | plotly
```
Comment (cldougl) [inline _posts/python/basic/table-subplots/2015-06-30-table-subplots.html]: rm this line 
it's a duplicate of the name argument below. To do this programatically you can make sure you're using the publisher cell format from the readme: https://github.com/plotly/documentation/edit/source-design-merge/_posts/python/README.md

File: _posts/python/basic/table-subplots/2015-06-30-table-subplots.html [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+has_thumbnail: true
+thumbnail: table_subplots.jpg
+layout: user-guide
+name: Tables in Subplots
```
Comment (cldougl) [inline _posts/python/basic/table-subplots/2015-06-30-table-subplots.html]: Table and Chart Subplots

File: _posts/python/basic/table-subplots/2015-06-30-table-subplots.html [hunk 4]
```diff
@@ -0,0 +1,544 @@
...
+language: python
+title: Tables in Subplots | plotly
```
Comment (cldougl) [inline _posts/python/basic/table-subplots/2015-06-30-table-subplots.html]: Table and Chart Subplots | plotly

File: _posts/python/basic/table-subplots/2015-06-30-table-subplots.html [hunk 5]
```diff
@@ -0,0 +1,544 @@
...
+<div class="output">
+
+
+<div class="output_area">
+
+<div class="prompt"></div>
+
+
+<div class="output_subarea output_stream output_stdout output_text">
+<pre>Help on function create_table in module plotly.figure_factory._table:
```
Comment (cldougl) [inline _posts/python/basic/table-subplots/2015-06-30-table-subplots.html]: this should point to the table reference: https://plot.ly/python/reference/#table

File: _posts/python/basic/table-subplots/table-subplots.ipynb [hunk 6]
```diff
@@ -0,0 +1,578 @@
...
+    "py.iplot(fig1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "#### Table and Centered Plots"
```
Comment (cldougl) [inline _posts/python/basic/table-subplots/table-subplots.ipynb]: I would say Vertical: Vertical Table and Graph Subplot rather than centered.

**Comment by cldougl** [comment]
2 main things:
- this should only refer to go.Table (i.e. the plotly.js tables not figure factory tables). Thus it's important to note that these are available in [v 2.1.0](https://github.com/plotly/plotly.py/blob/master/CHANGELOG.md#210---2017-10-10)

- the chart spacing looks a bit off:

screen shot 2017-11-01 at 6 37 19 pm


screen shot 2017-11-01 at 6 37 33 pm


After those points and the above comments are addressed I'll take another quick look but should be pretty much good to go from there!

**Comment by Kully** [comment]
@cldougl all comments addressed

**Comment by cldougl** [comment]
nice- the charts look great now!!
Can you make the first table using the plotly.js tables rather than figure factory?
Also remove the ff.table help cell from the reference section:

screen shot 2017-11-02 at 1 20 13 pm

File: _posts/python/subplots/table-subplots/table-subplots.ipynb [hunk 7]
```diff
@@ -0,0 +1,544 @@
...
+   ],
+   "source": [
+    "from IPython.display import display, HTML\n",
+    "\n",
+    "display(HTML('<link href=\"//fonts.googleapis.com/css?family=Open+Sans:600,400,300,200|Inconsolata|Ubuntu+Mono:400,700\" rel=\"stylesheet\" type=\"text/css\" />'))\n",
+    "display(HTML('<link rel=\"stylesheet\" type=\"text/css\" href=\"http://help.plot.ly/documentation/all_static/css/ipython-notebook-custom.css\">'))\n",
+    "\n",
+    "! pip install git+https://github.com/plotly/publisher.git --upgrade\n",
+    "import publisher\n",
+    "publisher.publish(\n",
```
Comment (cldougl) [inline _posts/python/subplots/table-subplots/table-subplots.ipynb]: I think the publisher cell isn't quite right. I know it's confusing cause some of our older examples have things duplicated / incorrect but you can always refer to the [readme](https://github.com/plotly/documentation/tree/source-design-merge/_posts/python) to see the correct order:

```
from IPython.display import display, HTML

display(HTML('<link href="//fonts.googleapis.com/css?family=Open+Sans:600,400,300,200|Inconsolata|Ubuntu+Mono:400,700" rel="stylesheet" type="text/css" />'))
display(HTML('<link rel="stylesheet" type="text/css" href="http://help.plot.ly/documentation/all_static/css/ipython-notebook-custom.css">'))

! pip install git+https://github.com/plotly/publisher.git --upgrade
import publisher
publisher.publish(
    'your-tutorial-chart.ipynb', 'python/your-tutorial-chart/', 'Your Tutorial Chart',
    'How to make your-tutorial-chart plots in Python with Plotly.',
    title = 'Python Your Tutorial Chart | Plotly',
    has_thumbnail='true', thumbnail='thumbnail/your-tutorial-chart.jpg', 
    language='python', 
    # page_type='example_index', // note this is only if you want the tutorial to appear on the main page: plot.ly/python
    display_as='chart_type', order=2,
    ipynb= '~notebook_demo/1')
```

so your `title` is correct but the 3rd arg should be the name (i.e. just `Table and Chart Subplots`)

**Comment by Kully** [comment]
@cldougl all done!

**Comment by cldougl** [comment]
awesome!! good to 💃  after making sure the thumbnail is added:
screen shot 2017-11-03 at 2 03 03 pm

Thanks for adding the 1st table!! One thing to do (not blocking - you can merge before if you'd like!) is to try and improve the header a bit:

screen shot 2017-11-03 at 2 02 48 pm
ie the labels are mostly getting cut off and the unnamed is centered while the others are top oriented.

**Comment by cldougl** [comment]
actually- it looks like the subplot tables are missing some rows

**Comment by Kully** [comment]
@cldougl

**Comment by cldougl** [comment]
Cool - the 1st subplot looks great!! 
screen shot 2017-11-08 at 12 51 39 pm

The table in the 2nd subplot is still incorrect though:
screen shot 2017-11-08 at 12 51 50 pm

**Comment by Kully** [comment]
> The table in the 2nd subplot is still incorrect though:

The problems I see are:
- there are some NaN values
- only 4 rows

Any other things before I fix?

**Comment by cldougl** [comment]
@Kully the date column isn't correct. Probably related to the NaNs- looks like the data wasn't formatted correctly

**Comment by Kully** [comment]
@cldougl tables look good now.

**Comment by cldougl** [comment]
💃 💃

**Pull request merged**